servo-media-gstreamer-render-android does not compile

**Describe the bug:**

```
error[E0053]: method `to_vec` has an incompatible type for trait
  --> components/media/backends/gstreamer/render-android/lib.rs:27:25
   |
27 |     fn to_vec(&self) -> Result<VideoFrameData, ()> {
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `std::option::Option<VideoFrameData>`, found `Result<VideoFrameData, ()>`
   |
   = note: expected signature `fn(&GStreamerBuffer) -> std::option::Option<VideoFrameData>`
              found signature `fn(&GStreamerBuffer) -> Result<VideoFrameData, ()>`
help: change the output type to match the trait
   |
27 -     fn to_vec(&self) -> Result<VideoFrameData, ()> {
27 +     fn to_vec(&self) -> std::option::Option<VideoFrameData> {
```

Presumably the trait was refactored, and the android backend not adapted, since it's not tested in CI


**To Reproduce:**
```
❯ ./mach exec cargo build -p servo-media-gstreamer-render-android
```

**Platform:**
Specify OS, distribution, and hardware platform.

--- a/components/media/backends/gstreamer/render-android/lib.rs
+++ b/components/media/backends/gstreamer/render-android/lib.rs
@@ -24,17 +24,17 @@ struct GStreamerBuffer {
 }
 
 impl Buffer for GStreamerBuffer {
-    fn to_vec(&self) -> Result<VideoFrameData, ()> {
+    fn to_vec(&self) -> Option<VideoFrameData> {
         // packed formats are guaranteed to be in a single plane
         if self.frame.format() == gstreamer_video::VideoFormat::Rgba {
-            let tex_id = self.frame.texture_id(0).map_err(|_| ())?;
-            Ok(if self.is_external_oes {
+            let tex_id = self.frame.texture_id(0).ok()?;
+            Some(if self.is_external_oes {
                 VideoFrameData::OESTexture(tex_id)
             } else {
                 VideoFrameData::Texture(tex_id)
             })
         } else {
-            Err(())
+            None
         }
     }
 }
@@ -124,8 +124,8 @@ impl Render for RenderAndroid {
             };
         }
 
-        let buffer = sample.buffer_owned().ok_or_else(|| ())?;
-        let caps = sample.caps().ok_or_else(|| ())?;
+        let buffer = sample.buffer_owned()?;
+        let caps = sample.caps()?;
 
         let is_external_oes = caps
             .structure(0)
@@ -148,8 +148,7 @@ impl Render for RenderAndroid {
             }
         }
 
-        let frame =
-            gstreamer_gl::GLVideoFrame::from_buffer_readable(buffer, &info).or_else(|_| Err(()))?;
+        let frame = gstreamer_gl::GLVideoFrame::from_buffer_readable(buffer, &info).ok()?;
 
         if self.gst_context.lock().unwrap().is_some() {
             if let Some(sync_meta) = frame.buffer().meta::<gstreamer_gl::GLSyncMeta>() {

